Added Commands, Fixed Trunk/Frunk

boxystudio opened the pull request:
**New Commands:**

- Window Control
- Software Update (if available)
- Frunk Actuation

**Fixed Commands:**

- Trunk Actuation

boxystudio committed:
Added Commands, Fixed Trunk/Frunk

New Commands:
Window Control
Software Update (if available)
Frunk Actuation

Fixed Commands:
Trunk Actuation

corsair commented on src/StephanGroen/Tesla/Tesla.php:
```diff
@@ -89,6 +89,11 @@ public function wakeUp() : array
+    {
+        return $this->sendRequest('/command/schedule_software_update', [ 'offset_sec' => 0 ], 'POST')['response'];
+    }
+
```
The offset seconds should be able to be set as a variable and default to 0 (immediate), to allow for delayed installations.

boxystudio replied:
Good idea! Added a new commit with this change.

corsair commented on src/StephanGroen/Tesla/Tesla.php:
```diff
@@ -157,6 +162,11 @@ public function lockDoors() : array
+    {
+        return $this->sendRequest('/command/window_control', [ 'command' => $state, 'lat' => 0, 'lon' => 0 ], 'POST')['response'];
+    }
+
```
We want to ensure that the window control command can receive latitude and longitude variables (defaulting to 0), as this is required for "close" but ignored for "vent".

corsair commented:
Thank you for your contribution, @boxystudio! I left some comments regarding some changes that would likely be beneficial to ensure full usage of the commands added.

Let me know your thoughts!

boxystudio committed:
Added variable for "seconds" in the startSoftwareUpdate method

corsair commented:
@boxystudio just checking in! Awaiting the windowControl function change to allow for this to behave properly, otherwise the usage of "close" would not work without the vehicle's current latitude/longitude.

More context:
https://github.com/timdorr/tesla-api/commit/e8ea847d73592c7aedcea83a11a41e9486600141

boxystudio committed:
Updated windowControl

Now includes variables for $lat and $lon to support Closing windows (requires latitude and longitude values that are close enough to the location of the car)

boxystudio commented:
Pushed an update to resolve this issue.

corsair reviewed (commented):
Looks great! Thank you for making this change.

Kozzirago commented:
Does anybody know if that API still works on the new Model 3 Highland?
